xbmc/filesystem/DVDDirectory.h

```diff
@@ -0,0 +1,28 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "IFileDirectory.h"
+
+namespace XFILE
+{
+
+/*!
+ \brief Abstracts a DVD virtual directory (dvd://) which in turn points to the actual physical drive
+ */
+class CDVDDirectory : public IFileDirectory
+{
+public:
+  CDVDDirectory() = default;
+  ~CDVDDirectory() = default;
```

Comment:
IFileDirectory does not have a virtual ctor? Or did you forget to add override here?

Reply:
oh, see comment below :)

Reply:
It doesn't have a virtual ctor. Defining one breaks multiple parts of the code where we call IFileDirectory constructor directly.
Should I keep it as is or should I remove it?

Reply:
afaik constructors can never be virtual, maybe ksooo meant the destructor?